api/resolvers/wallet.js

```diff
@@ -545,7 +539,47 @@ export default {
 
   Invoice: {
     satsReceived: i => msatsToSats(i.msatsReceived),
-    satsRequested: i => msatsToSats(i.msatsRequested)
+    satsRequested: i => msatsToSats(i.msatsRequested),
+    item: async (invoice, args, { models, me }) => {
+      if (!invoice.actionId) return null
+      switch (invoice.actionType) {
+        case 'ITEM_CREATE':
+        case 'ZAP':
+        case 'DOWN_ZAP':
+        case 'POLL_VOTE':
+          return (await itemQueryWithMeta({
+            me,
+            models,
+            query: `
+              ${SELECT}
+              FROM "Item"
+              WHERE id = $1`
+          }, Number(invoice.actionId)))?.[0]
+        default:
+          return null
+      }
+    },
+    itemAct: async (invoice, args, { models, me }) => {
+      const action2act = {
+        ZAP: 'TIP',
+        DOWN_ZAP: 'DONT_LIKE_THIS',
+        POLL_VOTE: 'POLL'
+      }
+      switch (invoice.actionType) {
+        case 'ZAP':
+        case 'DOWN_ZAP':
+        case 'POLL_VOTE':
+          return (await models.$queryRaw`
+              SELECT id, act, "invoiceId", "invoiceActionState", msats
+              FROM "ItemAct"
+              WHERE "ItemAct"."invoiceId" = ${Number(invoice.id)}
+              AND "ItemAct"."userId" = ${me?.id}
+              AND act = ${action2act[invoice.actionType]}::"ItemActType"`
+          )?.[0]
```

Comment:
```suggestion
          return (await models.$queryRaw`
              SELECT id, act, "invoiceId", "invoiceActionState", msats
              FROM "ItemAct"
              WHERE "ItemAct"."invoiceId" = ${Number(invoice.id)}::INTEGER
              AND "ItemAct"."userId" = ${me?.id}::INTEGER
              AND act = ${action2act[invoice.actionType]}::"ItemActType"`
          )?.[0]
```

Why `$queryRaw` though? Wouldn't `models.itemAct` work?

Reply:
Probably!